### PR TITLE
fix: sonar issues

### PR DESCRIPTION
**Pull request by daniellevass-bitrise**
Originally we started with the following issues:

Screenshot 2021-05-06 at 09 49 05

We end with the following issues:
Screenshot 2021-05-06 at 12 00 29

note: 476d0bf will cause a conflict with #189 - so this will need to be updated when that is merged first.

I fixed each issue with an individual commit and description - so it might help to read those for context too.

### CODE REVIEW

File: trace-sdk/src/androidTest/java/io/bitrise/trace/data/management/DataManagerInstrumentedTest.java [hunk 1]
```diff
@@ -204,16 +209,25 @@ public void reset_shouldNotBeInitialised() {
...
-     * converted to a {@link io.opencensus.proto.metrics.v1.Metric} it should not throw an exception.
+     * converted to a {@link io.opencensus.proto.metrics.v1.Metric} it should not throw an exception and be saved
+     * to the database correctly.
      */
     @Test
-    @UiThreadTest
     public void handleReceivedData_shouldHandleMetricWithoutException() {
         DataManager.getInstance(context);
         ApplicationSessionManager.getInstance().startSession();
+        final TraceDatabase traceDatabase = Room.inMemoryDatabaseBuilder(context, TraceDatabase.class).build();
```
Comment (richard-bogdan-bitrise) [inline trace-sdk/src/androidTest/java/io/bitrise/trace/data/management/DataManagerInstrumentedTest.java:219]: Did you know there is [https://github.com/bitrise-io/trace-android-sdk/blob/main/trace-sdk/src/androidTest/java/io/bitrise/trace/data/storage/TestDataStorage.java](https://github.com/bitrise-io/trace-android-sdk/blob/main/trace-sdk/src/androidTest/java/io/bitrise/trace/data/storage/TestDataStorage.java)? Or is it deliberate you did not use that?

File: trace-sdk/src/androidTest/java/io/bitrise/trace/data/management/DataManagerInstrumentedTest.java [hunk 2]
```diff
@@ -204,16 +209,25 @@ public void reset_shouldNotBeInitialised() {
 
     /**
      * When the {@link DataManager#handleReceivedData(Data)} is called with a valid {@link Data} that will be
-     * converted to a {@link io.opencensus.proto.metrics.v1.Metric} it should not throw an exception.
+     * converted to a {@link io.opencensus.proto.metrics.v1.Metric} it should not throw an exception and be saved
+     * to the database correctly.
      */
     @Test
-    @UiThreadTest
```
Comment (richard-bogdan-bitrise) [inline trace-sdk/src/androidTest/java/io/bitrise/trace/data/management/DataManagerInstrumentedTest.java:210]: Why did you remove this annotation?

File: trace-sdk/src/test/java/io/bitrise/trace/data/ULIDTest.java [hunk 3]
```diff
@@ -36,44 +38,44 @@ public void isValid_shouldBeValid() {
         assertThat(actualValue, is(true));
     }
 
+    /**
+     * Asserts that the provided String ulid is NOT valid.
+     * @param ulid the String to validate.
```
Comment (richard-bogdan-bitrise) [inline trace-sdk/src/test/java/io/bitrise/trace/data/ULIDTest.java:43]: nit: Please use the auto-formatting.

File: trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/BaseDataFormatter.java [hunk 4]
```diff
@@ -10,7 +10,7 @@
  * This ensures the ApplicationSessionManager is started before which is required to be able to
  * format any data
```
Comment (richard-bogdan-bitrise) [inline trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/BaseDataFormatter.java:11]: nit: I just spotted that there is no line ending dot, could you please add it there?

File: trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/BaseDataFormatter.java [hunk 5]
```diff
@@ -10,7 +10,7 @@
...
  */
-public class BaseDataFormatterTest {
+public class BaseDataFormatter {
```
Comment (richard-bogdan-bitrise) [inline trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/BaseDataFormatter.java:13]: I see your point, but I think this way it is much more confusing, because you would expect some function in it, based on it, but this is for having the common code for test classes. I would keep the original name, and instead make the class abstract, what do you think?

File: trace-test-application/src/androidTest/java/io/bitrise/trace/testapp/ActivityUiTest.java [hunk 6]
```diff
@@ -25,7 +27,7 @@
     @Test
     public void activityStateTest() {
         final IndexActivityScreen indexActivityScreen = new IndexActivityScreen(uiDevice);
-
+        assertNotNull(indexActivityScreen.getButtonUiTestsLabel());
```
Comment (richard-bogdan-bitrise) [inline trace-test-application/src/androidTest/java/io/bitrise/trace/testapp/ActivityUiTest.java:30]: I was wondering if we should add assertions for the app launch (launchTraceTestApp method) as this test is for the state of the activities of the app, not about the UI.

**Comment by richard-bogdan-bitrise** [comment]
Added some questions, please check them and let me know your thoughts.

File: trace-sdk/src/androidTest/java/io/bitrise/trace/data/management/DataManagerInstrumentedTest.java (hunk 2 above)
Comment (daniellevass-bitrise) [reply trace-sdk/src/androidTest/java/io/bitrise/trace/data/management/DataManagerInstrumentedTest.java:210]: Because I've updated the assertion to check the item now exists in the database, room won't work on the main thread. It gives this exception:
Screenshot 2021-05-10 at 19 11 47

Is there a reason you think it should be there?

File: trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/BaseDataFormatter.java (hunk 4 above)
Comment (daniellevass-bitrise) [reply trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/BaseDataFormatter.java:11]: Good catch. Added :)

File: trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/BaseDataFormatter.java (hunk 5 above)
Comment (daniellevass-bitrise) [reply trace-sdk/src/test/java/io/bitrise/trace/data/management/formatter/BaseDataFormatter.java:13]: Ooh that's a good suggestion. Will update the code and check to make sure Sonar is happy with it too.

File: trace-test-application/src/androidTest/java/io/bitrise/trace/testapp/ActivityUiTest.java (hunk 6 above)
Comment (daniellevass-bitrise) [reply trace-test-application/src/androidTest/java/io/bitrise/trace/testapp/ActivityUiTest.java:30]: Interesting - can you explain a bit more what you mean by that?

Comment (richard-bogdan-bitrise) [reply trace-test-application/src/androidTest/java/io/bitrise/trace/testapp/ActivityUiTest.java:30]: So method launchTraceTestApp returns a Boolean value if the launch was successful or not, we could add assertion for that method.

File: trace-sdk/src/androidTest/java/io/bitrise/trace/data/management/DataManagerInstrumentedTest.java (hunk 2 above)
Comment (daniellevass-bitrise) [reply trace-sdk/src/androidTest/java/io/bitrise/trace/data/management/DataManagerInstrumentedTest.java:210]: Ah interesting problem. That explains why the test occasionally fails on CI. I've added the annotation back so that the test runs on it's own - but made the actual work on an `AsyncTask`. I'll add a comment to about why this works this way for future. Have added this in a0e7e82 .

File: trace-test-application/src/androidTest/java/io/bitrise/trace/testapp/ActivityUiTest.java (hunk 6 above)
Comment (daniellevass-bitrise) [reply trace-test-application/src/androidTest/java/io/bitrise/trace/testapp/ActivityUiTest.java]: Ah. Sure. I've updated ActivityUiTest to use this - however the other tests (ApplicationStartUpUiTest and FragmentUiTest) will still use the checking the button exists. Does that sound okay?

Comment (richard-bogdan-bitrise) [reply trace-test-application/src/androidTest/java/io/bitrise/trace/testapp/ActivityUiTest.java]: Sure.

**Review by richard-bogdan-bitrise — APPROVED** [review]
Thanks for the updates, LGTM! 🚀
